javascript/net/grpc/web/generator/grpc_generator.cc

```diff
@@ -825,11 +825,11 @@ void PrintProtoDtsMessage(Printer* printer, const Descriptor* desc,
                      "set$js_field_name$(value?: $js_field_type$): "
                      "$class_name$;\n");
     }
-    if (field->type() == FieldDescriptor::TYPE_MESSAGE &&
+    if ((field->type() == FieldDescriptor::TYPE_MESSAGE || field->has_optional_keyword()) &&
         !field->is_repeated() && !field->is_map()) {
```

Comment:
nit: i'd probably prefer to maybe rewrite the logic as follows:

```
(field->has_optional_keyword() ||
    (field->type() == FieldDescriptor::TYPE_MESSAGE && 
        !field->is_repeated() &&
        !field->is_map())
```

Probably doesn't make a huge difference, but it seems to make a little bit more sense to me.. :P

WDYT? :)

Reply:
I'm not sure.

The semantics of the two conditions is only the same when you factor in the [proto3 language spec](https://github.com/protocolbuffers/protobuf/blob/master/docs/field_presence.md#presence-in-proto3-apis) disallowing `optional` on map or repeated fields.

I read it as _when the field is a message or optional, but not when repeated or a map_ as opposed to _when the field is optional (which implies it's not repeated or a map), or it's a message but not repeated or a map_.

I'll leave the final decision up to you though :)

Reply:
Matt - just a reminder about this comment :)

Reply:
> The semantics of the two conditions is only the same when you factor in the proto3 language spec disallowing optional on map or repeated fields.

Yeah i think that was my exactly concern.. -- that `optional` is mutually exclusive with `repeated` and `map` so i don't think they should be in a `AND` relationship.. :)

> I read it as when the field is a message or optional, but not when repeated or a map as opposed to when the field is optional (which implies it's not repeated or a map), or it's a message but not repeated or a map.

Yeah right.. although i felt that's a little long-winded.. (and i had some challenge understanding that too) 😅

Where i was thinking as: _when the field is optional, or a message that is not repeated or a map_.

I feel the latter is simpler and easier to reason. :)

If you are ok both ways, do you mind making the change? Thanks! :)

Reply:
The change has been made

Not sure what the process is around resolving conversations, so I'll leave this thread here

Reply:
Thanks a lot for the change! I guess it doesn't hurt to just leave them open here :D